go.sum

```diff
@@ -410,6 +410,7 @@ github.com/stretchr/testify v1.2.2/go.mod h1:a8OnRcib4nhh0OaRAV+Yts87kKdq0PP7pXf
 github.com/stretchr/testify v1.3.0 h1:TivCn/peBQ7UY8ooIcPgZFpTNSz0Q2U6UrFlUfqbe0Q=
 github.com/stretchr/testify v1.3.0/go.mod h1:M5WIy9Dh21IEIfnGCwXGc5bZfKNJtfHm1UVUgZn+9EI=
 github.com/tarm/serial v0.0.0-20180830185346-98f6abe2eb07/go.mod h1:kDXzergiv9cbyO7IOYJZWg1U88JhDg3PB6klq9Hg2pA=
+github.com/ugorji/go/codec v0.0.0-20181022190402-e5e69e061d4f/go.mod h1:VFNgLljTbGfSG7qAOspJ7OScBnGdDN/yBr0sguwnwf0=
 github.com/ugorji/go/codec v0.0.0-20181204163529-d75b2dcb6bc8/go.mod h1:VFNgLljTbGfSG7qAOspJ7OScBnGdDN/yBr0sguwnwf0=
 github.com/ugorji/go/codec v0.0.0-20181209151446-772ced7fd4c2 h1:EICbibRW4JNKMcY+LsWmuwob+CRS1BmdRdjphAm9mH4=
 github.com/ugorji/go/codec v0.0.0-20181209151446-772ced7fd4c2/go.mod h1:VFNgLljTbGfSG7qAOspJ7OScBnGdDN/yBr0sguwnwf0=
```

Comment:
Do you know, why there are diffierent versions for go codec listed here? Shouldn't only one be used?

Reply:
From `go help modules`
```
The go command maintains, in the main module's root directory alongside
go.mod, a file named go.sum containing the expected cryptographic checksums
of the content of specific module versions. Each time a dependency is
used, its checksum is added to go.sum if missing or else required to match
the existing entry in go.sum.
```

So according to my acknoledge, go will add all known versions of a used package in the module cache to `go.sum`, so the cryptographic signature of the package would be more "distributed" across every project. Running `go mod tidy` will not remove them even if I deleted the whole `go.sum` and generate it again.

Reply:
I think deleting my module cache and tidying again might solve the problem but it would still appear again in future invocations of `go mod tidy` by me or other contributors.

Reply:
I have another theory that the gzip package might require another version of `github.com/ugorji/go/codec`, but I since `go mod tidy` said we need this line let's just keep it

Reply:
https://github.com/gin-contrib/gzip/blob/master/go.sum#L22

I found this line, it was indeed overlaied from the gzip package. So we do need this line.

Reply:
Understood!